src/huggingface_hub/hf_api.py

```diff
@@ -506,6 +506,85 @@ def list_models(
         d = r.json()
         return [ModelInfo(**x) for x in d]
 
+    def list_models_from_filter(self, model_filter: ModelFilter):
```

Comment:
As you correctly guessed, I'm not a huge fan of offering another API to do the same thing - I'd rather this be merged in `list_models` with the support for the `ModelFilter` class.